pkg/agent/agent_test.go

```diff
@@ -188,24 +206,33 @@ func TestListReleases(t *testing.T) {
 				t.Errorf("got: %d, want: %d", got, want)
 			}
 
-			//Check if expected and returned contain the same releases, given they have same size
-			m := make(map[releaseStub]bool)
+			// Map a unique identifier to ptr to AppOverview
+			m := make(map[string]*proxy.AppOverview)
 
-			for _, eapp := range tc.expectedApps {
-				m[releaseStub{eapp.ReleaseName, eapp.Namespace, strtoi(eapp.Version)}] = true
+			for i, eapp := range tc.expectedApps {
+				m[getAppIdentity(eapp)] = &tc.expectedApps[i]
 			}
 
+			// All attained apps, must have a unique id that is already in the map
+			// Attained app and its mapping should be equal in structure not only identifier
 			for _, app := range apps {
-				rs := releaseStub{app.ReleaseName, app.Namespace, strtoi(app.Version)}
-				if _, ok := m[rs]; !ok {
-					t.Errorf("got: %v, want: %v", rs, "None")
+				appIdentity := getAppIdentity(app)
+				if expectedApp, ok := m[appIdentity]; !ok {
+					t.Errorf("got: %v, want: %v", &app, "None")
+				} else if !cmp.Equal(*expectedApp, app) {
+					t.Errorf(cmp.Diff(*expectedApp, app))
 				}
 			}
```

Comment:
you can just do:

```
			if !cmp.Equal(tc.expectedApps, apps) {
				t.Errorf(cmp.Diff(tc.expectedApps, apps))
			}
```

If you reorder the expected apps

Reply:
Sure thing. It can be done that way with the pair `app.ReleaseName` and `app.Namespace` as keys to sort on.

Did I understand correctly?

Reply:
nope, I meant that if you reorder the apps in `tc.expectedApps` to match the `apps` you don't need the map nor the iteration over the apps, you can just compare the arrays